google/cloud/documentai_toolbox/wrappers/document.py

```diff
@@ -481,6 +481,35 @@ def from_gcs(
             gcs_input_uri=gcs_input_uri,
         )
 
+    @classmethod
+    def list_from_gcs(
+        cls: Type["Document"],
+        gcs_bucket_name: str,
+        gcs_prefix: str,
+    ) -> List["Document"]:
+        r"""Loads a list of Documents from Cloud Storage.
+
+        Args:
+            gcs_bucket_name (str):
+                Required. The gcs bucket.
+
+                Format: Given `gs://{bucket_name}/{optional_folder}/{operation_id}/` where `gcs_bucket_name={bucket_name}`.
+            gcs_prefix (str):
+                Required. The prefix to the location of the target folder.
+
+                Format: Given `gs://{bucket_name}/{optional_folder}/{target_folder}` where `gcs_prefix={optional_folder}/{target_folder}`.
+        Returns:
+            List[Document]:
+                A List of documents from gcs.
+        """
+        return [
```

Comment:
This is assuming that every sub-directory in the parent (bucket+prefix) is acceptable by `Document.from_gcs`, which I think has some validation checks.

What is the desired behavior here if some of those sub-directories causes `Document.from_gcs` to fail?  Should the whole `list_from_gcs` call fail, or should it return just whatever is successfully loaded (maybe along with some metadata explaining the failed items)?

(Let's include unit tests for the case where one of the sub-directories contains wrong files.)

Reply:
`list_document_tree()` only outputs directories that contain JSON files. (However, it doesn't directly validate if they are `Document` JSON format.)

`.from_gcs()` will throw a `ValueError` if a file is invalid. Or the `documentai.Document.from_json()` method will throw an exception.

It would probably make sense for this batch case to return whatever was successfully loaded and just skip files that throw exceptions. (And printing out or returning the failed files.)